### PR TITLE
CA-286292: VLAN PIF on SR-IOV status is incorrect after restarting toolstack

### PR DESCRIPTION
**Pull request by krizex**
When synchronising PIF currently_attached, should ignore VLAN PIF on SR-IOV and
SR-IOV logical PIF because they don't have real vswitch bridge, so that the
bridge existance does not indicate the connection status of the PIF.

Signed-off-by: Yang Qian <yang.qian@citrix.com>

### CODE REVIEW

**Comment by mseri** [comment]
How do we decide if they are attached or not?

**Comment by krizex** [comment]
The status will be updated every time plug/unplug the PIF. Xapi restarting should not change the status of SR-IOV logical PIF and VLAN PIF on SR-IOV.

File: ocaml/xapi/xapi_host.ml [hunk 1]
```diff
@@ -1679,8 +1679,12 @@ let sync_pif_currently_attached ~__context ~host ~bridges =
   let networks = Db.Network.get_all_records ~__context in
   let pifs = Db.PIF.get_records_where ~__context ~expr:(
       Eq (Field "host", Literal (Ref.string_of host))
-    ) in
-
+    ) |> List.filter (fun (pif, pif_rec) ->
```
Comment (mseri) [inline ocaml/xapi/xapi_host.ml]: If you write `_pif` you'll avoid a warning for the unused `pif` variable

Comment (krizex) [reply ocaml/xapi/xapi_host.ml]: I'd like to replace it with `_`

**Comment by cheng-z** [comment]
Looks good to me also.  Merge

**Pull request merged**